File: app/src/main/java/com/abishov/hexocat/common/picasso/PicassoServiceLocator.java
package com.abishov.hexocat.common.picasso;

import android.content.Context;

import com.squareup.picasso.Picasso;
import com.squareup.picasso.Transformation;

public final class PicassoServiceLocator {
    private static String SERVICE_PICASSO = "service:picasso";
    private static String SERVICE_PICASSO_TRANSFORMATION = "service:picassoTransformation";

    private PicassoServiceLocator() {
        throw new AssertionError();
    }

    public static boolean matchesService(String name) {
        return SERVICE_PICASSO.equals(name);
    }

    public static boolean matchesTransformationService(String name) {
        return SERVICE_PICASSO_TRANSFORMATION.equals(name);
    }

    @SuppressWarnings({"ResourceType", "WrongConstant"})
    public static Picasso obtain(Context context) {
        return (Picasso) context.getSystemService(SERVICE_PICASSO);
    }

    @SuppressWarnings({"ResourceType", "WrongConstant"})
    public static Transformation obtainTransformation(Context context) {
        return (Transformation) context.getSystemService(SERVICE_PICASSO_TRANSFORMATION);
    }
}

File: app/src/main/java/com/abishov/hexocat/common/picasso/PicassoServiceLocator.kt
package com.abishov.hexocat.common.picasso

import android.content.Context
import com.squareup.picasso.Picasso
import com.squareup.picasso.Transformation

private const val SERVICE_PICASSO = "service:picasso"
private const val SERVICE_PICASSO_TRANSFORMATION = "service:picassoTransformation"

object PicassoServiceLocator {

  fun matchesService(name: String) = SERVICE_PICASSO == name

  fun matchesTransformationService(name: String) = SERVICE_PICASSO_TRANSFORMATION == name

  @SuppressWarnings("ResourceType", "WrongConstant")
  fun obtain(context: Context) = context.getSystemService(SERVICE_PICASSO) as Picasso

  @SuppressWarnings("ResourceType", "WrongConstant")
  fun obtainTransformation(context: Context): Transformation {
    return context.getSystemService(SERVICE_PICASSO_TRANSFORMATION) as Transformation
  }
}
